src/riak_kv_app.erl

```diff
@@ -49,7 +49,21 @@
 start(_Type, _StartArgs) ->
     riak_core_util:start_app_deps(riak_kv),
 
-    FSM_Limit = app_helper:get_env(riak_kv, fsm_limit, ?DEFAULT_FSM_LIMIT),
+    FSM_Limit = case app_helper:get_env(riak_kv, fsm_limit, ?DEFAULT_FSM_LIMIT) of
+                    undefined ->
+                        %% If it's explicitly set to 'undefined', that means "no limit".
+                        %% sidejob doesn't have an option to specify "infinity" but if
+                        %% we make the limit equal to the node's global process limit,
+                        %% then we're pretty much guaranteed to never exceed that:
+                        erlang:system_info(process_limit);
```

Comment:
@angrycub would it be wise to do something like 90% of the process limit, vs. all of it? Or subtract some minimal number (1k? 5k?) to reserve so if things go really south we're still providing some protection to the node?